elastic4s-core/src/main/scala/com/sksamuel/elastic4s/requests/indexes/MappingHandlers.scala

```diff
@@ -28,12 +29,32 @@ trait MappingHandlers {
       }
     }
 
+    private def getMappings(mappings: Map[String, Any]) = {
+      mappings
+        .getOrElse("properties", getFieldMappings(mappings))
+    }
+
+    private def getFieldMappings(possibleFieldMapping: Map[String, Any]) = {
+      possibleFieldMapping.size match {
+        case 0 => Map.empty
+        case _: Int => possibleFieldMapping.head._2 match {
+          case map: Map[String, Any] => map.getOrElse("mapping", Map.empty)
+          case _ => Map.empty
```

Comment:
This will match when there are no props, but it's not empty. E.g.

```
{
    "indexnoprops": {
        "mappings": {
            "dynamic": "strict"
        }
    }
}
```